### PR TITLE
Add RPC call and config option to rotate keys without restarting

### PR DESCRIPTION
**Pull request by fiaxh**
## Pull request checklist

- [ ] All tests pass. The project builds and runs.
- [ ] I have resolved any merge conflicts.
- [ ] I have resolved all `clippy` and `rustfmt` warnings.

### CODE REVIEW

File: validator/src/key_utils.rs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+    pub fn get_keys(&self) -> Vec<BlsKeyPair> {
+        self.keys.values().cloned().collect()
+    }
+
+    pub fn update_current_key(&mut self, public_key: &CompressedPublicKey) {
+        self.current_key = self
+            .keys
+            .get(public_key)
+            .expect("Invalid validator configuration: Expected voting key not available")
```
Comment (hrxi) [inline validator/src/key_utils.rs]: This is the user-visible panic when the user starts the client without the correct key, but also when the key is rotated on-chain and an election block happens.

I'm not sure what the wanted behavior is here. It's probably good to quit the client if the correct signing key isn't available on startup, but I'm not sure what to do when the client is already running and has hit an election block.

File: validator/src/validator.rs [hunk 2]
```diff
@@ -70,7 +71,7 @@ struct InactivityState {
 pub struct ValidatorProxy {
     pub validator_address: Arc<RwLock<Address>>,
     pub signing_key: Arc<RwLock<SchnorrKeyPair>>,
-    pub voting_key: Arc<RwLock<BlsKeyPair>>,
+    pub voting_key: Arc<RwLock<VotingKeys>>,
```
Comment (hrxi) [inline validator/src/validator.rs]:
```suggestion
    pub voting_keys: Arc<RwLock<VotingKeys>>,
```
Here and below.

**Review by hrxi — COMMENTED** [review]
It would be nice if the same was available for signing keys as these can also be rotated.

**Comment by fiaxh** [comment]
> It would be nice if the same was available for signing keys as these can also be rotated.

Yes, I agree. The implementations will look very similar. Thus, I wanted to first get this reviewed, such that I'm not doing every mistake twice, and then have to fix everything twice :slightly_smiling_face:

File: validator/src/validator.rs [hunk 3]
```diff
@@ -153,7 +154,7 @@ where
         validator_address: Address,
         automatic_reactivate: bool,
         signing_key: SchnorrKeyPair,
-        voting_key: BlsKeyPair,
+        voting_key: VotingKeys,
```
Comment (hrxi) [inline validator/src/validator.rs]:
```suggestion
        voting_keys: VotingKeys,
```

**Review by hrxi — COMMENTED** [review]
Looks good to me.

**Comment by hrxi** [comment]
Rebased for merging.

**Comment by sisou** [comment]
@fiaxh Ping on the version of this for signing keys, as talked about above.

**Pull request merged**